### DISCUSSION TITLE
How to define the number of HTTPFS threads for parallel data fetching?

### DISCUSSION THREAD

**Discussion by dforsber** [duckdb/duckdb#1766] [Q&A]
Noticed that you have added a very nice httpfs extension, including support for S3. 
- Is there a possibility to configure number of concurrent connections / IO threads for the extension?
- If yes, how would the IO threads be used to load balance data transfers?

**Comment by adriangb** [comment]
I'm also interested in this. I'm trying to query a large dataset in S3 and it looks like most of the time is spent doing IO. I'm interested in knowing:
- what's the default number of threads/concurrency used?
- is this configurable?
- will multiple threads/connections read byte ranges from a single file or only multiple files (i.e. would it help to split up a single large file into multiple smaller ones)
- could writing to S3 also be done in parallel?
